VMEncryption/main/oscrypto/ubuntu_1604/encryptstates/PatchBootSystemState.py

```diff
@@ -146,22 +148,44 @@ def _modify_pivoted_oldroot(self):
             raise Exception(message)
         else:
             self.context.logger.log("Patch found at path: {0}".format(patchpath))
-        
+
         self.command_executor.ExecuteInBash('patch -b -d /usr/share/initramfs-tools/hooks -p1 <{0}'.format(patchpath), True)
 
         os_volume = None
-        if os.path.exists(CommonVariables.az_symlink_os_volume) and os.path.realpath(CommonVariables.az_symlink_os_volume) == os.path.realpath(self.rootfs_block_device):
-            os_volume = CommonVariables.az_symlink_os_volume
+        az_symlink_os_volume = self._get_az_symlink_os_volume()
+        if os.path.exists(az_symlink_os_volume) and os.path.realpath(az_symlink_os_volume) == os.path.realpath(self.rootfs_block_device):
+            os_volume = az_symlink_os_volume
         else:
             os_volume = self.rootfs_block_device
-        
+
         entry = 'osencrypt {0} none luks,discard,header=/boot/luks/osluksheader,keyscript=/usr/sbin/azure_crypt_key.sh'.format(os_volume)
         self._append_contents_to_file(entry, '/etc/crypttab')
 
         self.command_executor.Execute('update-initramfs -u -k all', True)
         self.command_executor.Execute('update-grub', True)
         self.command_executor.Execute('grub-install --recheck --force {0}'.format(self.rootfs_disk), True)
 
+    def _get_az_symlink_os_volume(self):
```

Comment:
Can we move it to oscrypto?...That way both Ubuntu distros can call it

Reply:
Done